src/compiler/transformers/es2015.ts

```diff
@@ -3337,8 +3337,12 @@ namespace ts {
         scoped: false,
         priority: 0,
         text: `
+            var __extendStatics = (this && this.__extendStatics) ||
+                Object.setPrototypeOf ||
+                ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
+                function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };

```

Comment:
good point. then we do not need the `(this && this.__extendStatics)` part?

You can always override the `__extends` behavior, so it is not that we are restricting access t this functionality.